src/transformers/pipelines/fill_in_middle.py

```diff
@@ -0,0 +1,333 @@
+    objective, which includes the uni-directional models in the library (e.g. gpt2). See the list of available models
+    on [huggingface.co/models](https://huggingface.co/models?filter=text-generation).
+    """
+
+    # There are two modes in which infilling is supported: PSM (Prefix-Suffix-Middle) and SPM (Suffix-Prefix-Middle)
+    # CodeLlama's HF implementation has a third mode called 'suffix-first' which is added here for total compatibility with CodeLlama family
+    # The placement of prefix and suffix along with their respective sentinel tokens depends on the mode
+    # More information on this: https://arxiv.org/abs/2207.14255
+    DEFAULT_INFILL_MODE = "psm"
+    DEFAULT_INFILL_TOKEN = "<FILL_ME>"
+    ALL_INFILL_MODES = ["psm", "spm", "suffix-first"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Check if the provided model is among the supported ones
+        if self.model.name_or_path not in SUPPORTED_MODELS:
+            raise ValueError(f"The model '{self.model.name_or_path}' is not supported for the Fill-in-Middle task.")
+
+        # Get the Prefix, Suffix and Middle tokens for the corresponding model
+        self.PREFIX_TOKEN, self.MIDDLE_TOKEN, self.SUFFIX_TOKEN = SUPPORTED_MODELS[self.model.name_or_path]
+
+        self.check_model_type(
+            TF_MODEL_FOR_CAUSAL_LM_MAPPING_NAMES if self.framework == "tf" else MODEL_FOR_CAUSAL_LM_MAPPING_NAMES
+        )
+        if "prefix" not in self._preprocess_params:
+            # This is very specific. The logic is quite complex and needs to be done
+            # as a "default".
+            # It also defines both some preprocess_kwargs and generate_kwargs
+            # which is why we cannot put them in their respective methods.
+            prefix = None
+            if self.model.config.prefix is not None:
+                prefix = self.model.config.prefix
+
+            if prefix is not None:
+                # Recalculate some generate_kwargs linked to prefix.
+                preprocess_params, forward_params, _ = self._sanitize_parameters(prefix=prefix, **self._forward_params)
+                self._preprocess_params = {
+                    **self._preprocess_params,
+                    **preprocess_params,
+                }
+                self._forward_params = {**self._forward_params, **forward_params}
+
+    def ensure_infill_token(self, prompt_text, infill_token):
+        """
+        Ensures the prompt text has an infill token in it.
+        """
+        if infill_token not in prompt_text:
+            raise ValueError(f"Infill token '{infill_token}' not in the prompt")
+
+    def extract_prefix_suffix(self, input_text, infill_token):
+        """
+        Extracts the prefix and suffix from the input text.
+        """
+        prefix, _, suffix = input_text.partition(infill_token)
+        return prefix, suffix
```

Comment:
we probably don't need that one liner

Reply:
Are you suggesting that instead of making a function for extracting prefix and suffix, I just do it in-line? Because we need this functionality to extract the prefix and suffix from the user's provided prompt.

Reply:
yes 👍🏻